web/src/typeahead_helper.ts

```diff
@@ -112,6 +115,30 @@ export let render_person = (person: UserPillData | UserOrMentionPillData): strin
 
     const pronouns = pronouns_list?.[0]?.value;
 
+    // Prefer showing a matching custom profile field over email when the query matches it.
```

Comment:
We are showing custom profile field only if email does not matches. I am not sure what is the expected behavior if both match. This comment as of now feels like we would show custom field over email if both matches but that is not the case.